src/translator/response_builder.cpp

```diff
@@ -19,7 +19,7 @@ void ResponseBuilder::buildQualityScores(Histories &histories,
     // logprobs.
     auto normalizedPathScore = std::get<2>(result);
     auto wordQualities = hyp->tracebackWordScores();
-    wordQualities.pop_back();
+    wordQualities.pop_back(); //remove EOS token
```

Comment:
Try to not pop EOS and also return a quality score for EOS in your pipeline. It should be possible to provide quality score to EOS token.

The `pop_back()` needs to happen after your transformation for now, not before it goes into your pipeline.

Reply:
@jerinphilip indeed. Thanks for the notice.